scripts/build_char2array.py

```diff
@@ -0,0 +1,105 @@
+def has_glyph(font: TTFont, glyph: str) -> bool:
+    assert len(glyph) == 1, glyph
+    for table in font["cmap"].tables:
+        if ord(glyph) in table.cmap.keys():
+            return True
+    return False
+
+
+# Find correct font that supports symbol or returns None
+def pick_font(symbol: str, fonts: list) -> str | None:
+    for font in fonts:
+        if has_glyph(TTFont(font), symbol):
+            return font
+    return None
+
+
+def build_char2array(
+    save_to: str,
+    fonts_directory: str,
+    chars_path: str,
+    font_size: int = 16,
+    fill_value_unknown: int = 255,
+    width_unknown: int = None,
+    ignore_invalid_encoding: bool = False,
+):
+    assert fill_value_unknown in range(256), "Value of filling must be int number in [0,255]"
+    fonts = [f"{fonts_directory}/{font}" for font in listdir(fonts_directory) if font.endswith(("ttf", "otf"))]
+
+    with open(chars_path, "rb") as f:
+        all_chars = pickle.load(f)
+
+    char2array = dict()
+    not_supported = []
+    problem_chars = []
+    for char in tqdm(all_chars):
+        current_font = pick_font(char, fonts)
+
+        if current_font:
+            try:
+                char2array[char] = text2numpy(char, font=current_font, font_size=font_size)
+            except:
+                problem_chars.append(char)
+        else:
+            not_supported.append(char)
+
+    if not ignore_invalid_encoding:
+        for char in not_supported:
+            fixed_char, explanation_of_error = ftfy.fix_and_explain(char)
+            # if char was fixed and there is explanation of error (char encoding is broken)
+            if fixed_char != char and explanation_of_error:
+                try:
+                    current_font = pick_font(char, fonts)
+                    if current_font:
+                        char2array[char] = text2numpy(fixed_char, font=current_font, font_size=font_size)
+                except:
+                    problem_chars.append(char)
+                else:
+                    not_supported.remove(char)
+
+    not_supported += problem_chars
+    if not width_unknown:
+        width_unknown = int(np.median(np.array([v.shape[1] for k, v in char2array.items()])))
+    char2array["UNK"] = np.full(shape=(font_size, width_unknown), fill_value=fill_value_unknown, dtype=np.int_)
+
+    pickle.dump(char2array, open(save_to, "wb"))
+
+    logger.info(
+        f"Number of symbols in char2array: {len(char2array)}, len of not_supported symbols: {len(not_supported)}"
+    )
+    logger.info(f"Char2array covers {round(len(char2array)/len(all_chars)*100,2)}% of all symbols")
+
+    return char2array, not_supported
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--save-to", type=str, default="resources/char2array.pkl")
+    parser.add_argument("--fonts-directory", type=str, default="resources/fonts")
+    parser.add_argument("--chars-path", type=str, default="resources/data/chars.pkl")
```

Comment:
I guess it is better to store chars in the `.txt` format. So it is easy to check them and add new ones.